### PR TITLE
chore(udf): prepare for publishing java udf to maven

### PR DESCRIPTION
**Pull request by wangrunji0408**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Update POMs and publish the risingwave-udf library to the maven central repository.

- change group id from `com.risingwave.java` to  `com.risingwave`.
- rename root project to `risingwave-java-root`
- complete missing information for deployment: name, url, description, svn, developer info...
- publish `risingwave-java-root` and `risingwave-udf` v0.1.0 to the maven central repository.
- update versions to `0.1.0-SNAPSHOT`

https://s01.oss.sonatype.org/#nexus-search;quick~risingwave

## Checklist

- [ ] ~~I have written necessary rustdoc comments~~
- [ ] ~~I have added necessary unit tests and integration tests~~
- [ ] ~~I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).~~
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by wenym1** [comment]
I think we can publish all modules instead of only the udf ones to the maven repo every time we release a new version.

We are working on embedding jvm to compute node. If it works, the java udf may be able to run on the embedded jvm, and then it will somehow share the same jvm context with connector node. It has some benefits if they have the same parent pom and manage the dependencies together.

**Comment by wangrunji0408** [comment]
I see. But some of our users are requesting udf from maven recently. Can we publish the udf first and reattach it to parent again when the whole project is ready?

**Comment by liurenjie1024** [comment]
> I think we can publish all modules instead of only the udf ones to the maven repo every time we release a new version.
> 
> We are working on embedding jvm to compute node. If it works, the java udf may be able to run on the embedded jvm, and then it will somehow share the same jvm context with connector node. It has some benefits if they have the same parent pom and manage the dependencies together.

+1. But we can publish it first to solve customer's requirments

**Comment by wenym1** [comment]
I think the whole project can already be published directly, though no one will use other modules. 

On users' maven project the only the udf module should be added to dependency list and other modules is not touched.

**Comment by wenym1** [comment]
And I think running `mvn clean deploy --pl udf --am` will only publish the udf module.

**Comment by wangrunji0408** [comment]
Cool.

Btw, do you have any experience on publishing java libraries to maven central repository? I'm reading [the docs](https://central.sonatype.org/publish/publish-guide/#provide-files-checksums) and I find it much more complicated than publishing Rust libraries to crates.io.

**Comment by wangrunji0408** [comment]
I've registered the group id `com.risingwave` yesterday. https://issues.sonatype.org/browse/OSSRH-94627

Should we change the group id of these projects from `com.risingwave.java` to `com.risingwave`? I feel `.java` can be omitted in context of java development.

**Comment by liurenjie1024** [comment]
> I've registered the group id `com.risingwave` yesterday. https://issues.sonatype.org/browse/OSSRH-94627
> 
> Should we change the group id of these projects from `com.risingwave.java` to `com.risingwave`? I feel `.java` can be omitted in context of java development.

+1, I don't think we should add `.java`

**Review by liurenjie1024 — APPROVED** [review]

File: java/udf-example/pom.xml [hunk 1]
```diff
@@ -1,22 +1,16 @@
...
   <modelVersion>4.0.0</modelVersion>
 
-  <parent>
-    <artifactId>java-parent</artifactId>
-    <groupId>com.risingwave.java</groupId>
-    <version>1.0-SNAPSHOT</version>
-    <relativePath>../pom.xml</relativePath>
-  </parent>
-
   <groupId>com.example</groupId>
```
Comment (wenym1) [inline java/udf-example/pom.xml:8]: The group id should be `com.risingwave`?

**Review by wenym1 — APPROVED** [review]
Rest LGTM!

File: java/udf-example/pom.xml (hunk 1 above)
Comment (wangrunji0408) [reply java/udf-example/pom.xml:8]: This project is designed to be a template for users to quick start. So its group id could be anything.

Comment (wangrunji0408) [reply java/udf-example/pom.xml]: fixed. I'll create a new repo for the template.

**Pull request merged**